distributed/deploy/cluster.py

```diff
@@ -319,13 +319,20 @@ def logs(self, *args, **kwargs):
 
     @property
     def dashboard_link(self):
-        try:
-            port = self.scheduler_info["services"]["dashboard"]
-        except KeyError:
-            return ""
-        else:
-            host = self.scheduler_address.split("://")[1].split("/")[0].split(":")[0]
-            return format_dashboard_link(host, port)
+        if hasattr(self, "scheduler_spec"):
+            use_dashboard = self.scheduler_spec["options"]["dashboard"]
```

Comment:
What about `getattr` with a default return of an empty dictionary? Then you can check for the next two keys using `get()`.